<a href="https://colab.research.google.com/github/anlianguzova/BI-stat-course-2022/blob/olympic_games_branch/olympic_games_folder/report_folder/project_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import numpy as np
from scipy import stats

### 1. Function combining files

In [ ]:
def combine_files(path: str, extension: str='scv') -> pd.DataFrame:
    import glob
    suit_files = [name for name in glob.glob(fr'{path}*.*'.format(extension))]
    return pd.concat([pd.read_csv(file) for file in suit_files])

### Data loading

In [ ]:
path = 'data_folder/athlete_events/'
athletes = combine_files(path)
athletes.to_csv("data/athletes.csv", index=False, encoding='utf-8-sig') # if it is convenient to look for tiny flaws in the data using some csv viewer
athletes # or just open the combined file (better in a new tab)

ID                    Name Sex   Age  Height  Weight         Team  \
0      68370          Helmut Lehmann   M  25.0   178.0    77.0  Switzerland   
1      68371            Jens Lehmann   M  24.0   188.0    82.0      Germany   
2      68371            Jens Lehmann   M  24.0   188.0    82.0      Germany   
3      68371            Jens Lehmann   M  32.0   188.0    82.0      Germany   
4      68371            Jens Lehmann   M  32.0   188.0    82.0      Germany   
...      ...                     ...  ..   ...     ...     ...          ...   
22597  57367  Velik Nikolov Kapsazov   M  21.0   167.0    67.0     Bulgaria   
22598  57367  Velik Nikolov Kapsazov   M  21.0   167.0    67.0     Bulgaria   
22599  57367  Velik Nikolov Kapsazov   M  25.0   167.0    67.0     Bulgaria   
22600  57367  Velik Nikolov Kapsazov   M  25.0   167.0    67.0     Bulgaria   
22601  57367  Velik Nikolov Kapsazov   M  25.0   167.0    67.0     Bulgaria   

       NOC        Games    Year  Season         City       Sport  \
0      SUI  1984 Summer  1984.0  Summer  Los Angeles    Canoeing   
1      GER  1992 Summer  1992.0  Summer    Barcelona     Cycling   
2      GER  1992 Summer  1992.0  Summer    Barcelona     Cycling   
3      GER  2000 Summer  2000.0  Summer       Sydney     Cycling   
4      GER  2000 Summer  2000.0  Summer       Sydney     Cycling   
...    ...          ...     ...     ...          ...         ...   
22597  BUL  1956 Summer  1956.0  Summer    Melbourne  Gymnastics   
22598  BUL  1956 Summer  1956.0  Summer    Melbourne  Gymnastics   
22599  BUL  1960 Summer  1960.0  Summer         Roma  Gymnastics   
22600  BUL  1960 Summer  1960.0  Summer         Roma  Gymnastics   
22601  BUL  1960 Summer  1960.0  Summer         Roma  Gymnastics   

                                                Event   Medal  
0            Canoeing Men's Kayak Fours, 1,000 metres     NaN  
1      Cycling Men's Individual Pursuit, 4,000 metres  Silver  
2            Cycling Men's Team Pursuit, 4,000 metres    Gold  
3      Cycling Men's Individual Pursuit, 4,000 metres  Silver  
4            Cycling Men's Team Pursuit, 4,000 metres    Gold  
...                                               ...     ...  
22597                          Gymnastics Men's Rings     NaN  
22598                Gymnastics Men's Pommelled Horse     NaN  
22599          Gymnastics Men's Individual All-Around     NaN  
22600                Gymnastics Men's Team All-Around     NaN  
22601                                    Gymnastics M     NaN  

[271115 rows x 15 columns]

### 2. Data analysis and filtering

Something strange happened to the *Team* column: there are *Sao Tome and Principe*, *Socit Nautique de Bayonne*, *Pigoule* or countries with numbers!!!!!!! Let's look at it thoroughly

In [ ]:
athletes_upd = athletes.replace(to_replace='\-[0-9]*', value='', regex=True) #getting rid of counties with numbers
athletes_upd.Team.unique() # less values but still...

array(['Switzerland', 'Germany', 'East Germany', 'Australia', 'Finland',
       'Norway', 'Luxembourg', 'West Germany', 'Czechoslovakia',
       'Pigoule', 'France', 'Ukraine', 'Belgium', 'Antigua and Barbuda',
       'Austria', 'Denmark', 'Estonia', 'China', 'Hong Kong', 'Myanmar',
       'American Samoa', 'South Africa', 'Canada', 'United States',
       'Latvia', 'Argentina', 'Spain', 'Sweden', 'Iceland',
       'Great Britain', 'Netherlands', 'Hungary', 'Brazil', 'Portugal',
       'Sao Tome and Principe', 'New Zealand', 'Uruguay', 'Angola',
       'Italy', 'Venezuela', 'Guatemala', 'Greece',
       'Socit Nautique de Bayonne', 'Soviet Union', 'Slovenia',
       'Bulgaria', 'Yugoslavia', 'Kenya', 'Croatia', 'Poland', 'Belarus',
       'Zambia', 'Gabon', 'Japan', 'Phalainis ton Thorichtou "Hydra"',
       'Colombia', 'Cuba', 'Mexico', 'Trinidad and Tobago', 'Ireland',
       'Czech Republic', 'Philadelphia Turngemeinde, Philadelphia',
       'Syndi', 'Puerto Rico', 'Israel', 'Indone

There are toooooo many non-countries names, so i'm gonna filter it

In [ ]:
country_list = ['Afghanistan', 'Aland Islands', 'Albania', 'Algeria', 'American Samoa', 'Andorra', 'Angola', 'Anguilla', 'Antarctica', 'Antigua and Barbuda', 'Argentina', 'Armenia', 'Aruba', 'Australia', 'Austria', 'Azerbaijan', 'Bahamas', 'Bahrain', 'Bangladesh', 'Barbados', 'Belarus', 'Belgium', 'Belize', 'Benin', 'Bermuda', 'Bhutan', 'Bolivia, Plurinational State of', 'Bonaire, Sint Eustatius and Saba', 'Bosnia and Herzegovina', 'Botswana', 'Bouvet Island', 'Brazil', 'British Indian Ocean Territory', 'Brunei', 'Bulgaria', 'Burkina Faso', 'Burma', 'Burundi', 'Cambodia', 'Cameroon', 'Canada', 'Cape Verde', 'Cayman Islands', 'Central African Republic', 'Chad', 'Czechoslovakia', 'Chile', 'China', 'Christmas Island', 'Cocos (Keeling) Islands', 'Colombia', 'Comoros', 'Congo', 'Congo, The Democratic Republic of the', 'Congo (Brazzaville)', 'Congo (Kinshasa)', 'Cook Islands', 'Costa Rica', "Côte d'Ivoire", 'Croatia', 'Cuba', 'Curaçao', 'Cyprus', 'Czech Republic', 'Denmark', 'Djibouti', 'Dominica', 'Dominican Republic', 'East Germany', 'Ecuador', 'Egypt', 'El Salvador', 'England', 'Equatorial Guinea', 'Eritrea', 'Estonia', 'Ethiopia', 'Falkland Islands (Malvinas)', 'Faroe Islands', 'Fiji', 'Finland', 'France', 'French Guiana', 'French Polynesia', 'French Southern Territories', 'Gabon', 'Gambia', 'Georgia', 'Germany', 'Ghana', 'Gibraltar', 'Great Britain', 'Greece', 'Greenland', 'Grenada', 'Guadeloupe', 'Guam', 'Guatemala', 'Guernsey', 'Guinea', 'Guinea Bissau', 'Guyana', 'Haiti', 'Heard Island and McDonald Islands', 'Holy See (Vatican City State)', 'Honduras', 'Hong Kong', 'Hungary', 'Iceland', 'India', 'Indonesia', 'Iran, Islamic Republic of', 'Iraq', 'Ireland', 'Isle of Man', 'Israel', 'Italy', 'Jamaica', 'Japan', 'Jersey', 'Jordan', 'Kazakhstan', 'Kenya', 'Kiribati', "Korea, Democratic People's Republic of", 'Korea, Republic of', 'Kuwait', 'Kyrgyzstan', "Lao People's Democratic Republic", 'Latvia', 'Lebanon', 'Lesotho', 'Liberia', 'Libya', 'Liechtenstein', 'Lithuania', 'Luxembourg', 'Macao', 'Macedonia, Republic of', 'Madagascar', 'Malawi', 'Malaysia', 'Maldives', 'Mali', 'Malta', 'Marshall Islands', 'Martinique', 'Mauritania', 'Mauritius', 'Mayotte', 'Mexico', 'Micronesia, Federated States of', 'Moldova', 'Monaco', 'Mongolia', 'Montenegro', 'Montserrat', 'Morocco', 'Mozambique', 'Myanmar', 'Namibia', 'Nauru', 'Nepal', 'Netherlands', 'New Caledonia', 'New Zealand', 'Nicaragua', 'Niger', 'Nigeria', 'Niue', 'Norfolk Island', 'Northern Mariana Islands', 'North Korea', 'Norway', 'Oman', 'Pakistan', 'Palau', 'Palestine', 'Panama', 'Papua New Guinea', 'Paraguay', 'Peru', 'Philippines', 'Pitcairn', 'Poland', 'Portugal', 'Puerto Rico', 'Qatar', 'Réunion', 'Romania', 'Russia', 'Rwanda', 'Saint Barthélemy', 'Saint Helena, Ascension and Tristan da Cunha', 'Saint Kitts and Nevis', 'Saint Lucia', 'Saint Martin (French part)', 'Saint Pierre and Miquelon', 'Saint Vincent and the Grenadines', 'Samoa', 'San Marino', 'Sao Tome and Principe', 'Saudi Arabia', 'Senegal', 'Serbia', 'Serbia and Montenegro', 'Seychelles', 'Sierra Leone', 'Singapore', 'Sint Maarten (Dutch part)', 'Slovakia', 'Slovenia', 'Solomon Islands', 'Somalia', 'Soviet Union', 'South Africa', 'South Georgia and the South Sandwich Islands', 'Spain', 'Sri Lanka', 'Sudan', 'Suriname', 'South Korea', 'South Sudan', 'Svalbard and Jan Mayen', 'Swaziland', 'Sweden', 'Switzerland', 'Syria', 'Taiwan', 'Tajikistan', 'Tanzania, United Republic of', 'Thailand', 'Timor-Leste', 'Togo', 'Tokelau', 'Tonga', 'Trinidad and Tobago', 'Tunisia', 'Turkey', 'Turkmenistan', 'Turks and Caicos Islands', 'Tuvalu', 'Uganda', 'Ukraine', 'Unified Team', 'United Arab Emirates', 'United States', 'United States Minor Outlying Islands', 'United States Virgin Islands','Uruguay', 'Uzbekistan', 'Vanuatu', 'Venezuela', 'Vietnam', 'Virgin Islands, British', 'Virgin Islands', 'Wallis and Futuna', 'West Germany', 'Yemen', 'Yugoslavia', 'Zambia', 'Zimbabwe', 'Individual Olympic Athletes', 'South Vietnam', 'Iran', 'United Arab Republic', 'France/Great Britain', 'Australia/Great Britain', 'Scotland', 'North Yemen', 'Comoros', 'Oman', 'Tanzania', 'Holland', 'Macedonia', 'Hungaria', 'Italia', 'Kosovo', ]

In [ ]:
not_country = athletes_upd.query("Team not in @country_list")
not_country.Team.unique()

array(['Pigoule', 'Socit Nautique de Bayonne',
       'Phalainis ton Thorichtou "Hydra"',
       'Philadelphia Turngemeinde, Philadelphia', 'Syndi',
       'Pupilles de Neptune de Lille #1', 'Mascaret', "L'Aile VI",
       'Galopin', 'Sibindi', 'Pupilles de Neptune de Lille',
       'Toronto Argonauts', 'Deutscher Schwimm Verband Berlin',
       'Chinese Taipei', 'Nautikos Omilos', 'Jupiter',
       'Socialer Turnverein, Detroit', 'Tengiri', 'Pan',
       'Berliner Ruderverein von 1876', 'Malaya',
       'Chicago Athletic Association', 'Combine', 'Mohawk Indians',
       'Wiking', 'Ingegerd', 'Osborne Swimming Club, Manchester',
       'Netherlands Antilles', 'Marga', 'Lucky Girl', 'Nina', 'Xantippa',
       'Gallia II', 'Erna Signe', 'Galt Football Club', 'QuandMme',
       'Dolfijn', 'Mouchette', 'New College, Oxford', 'SagaII', 'Saga',
       'Vesper Boat Club', 'Thames Rowing Club', 'Moseley Wanderers',
       'Refugee Olympic Athletes', 'Norman', 'Boer Team',
       'Phalainis ton

In [ ]:
athletes_filtered = athletes_upd.loc[athletes_upd['Team'].isin(country_list)] # left only people from adequate countries. 

one can look for the names of people with strange teams and compare with other raws with this person but i don't have a mood for this....

There are also one strange value in Height column (340)

In [ ]:
athletes_filtered[athletes_filtered['Height'] == 340]

ID                            Name Sex   Age  Height  Weight  \
227  23549  Kirsty Leigh Coventry (Seward)   F  28.0   340.0    64.0   

         Team  NOC        Games    Year  Season    City     Sport  \
227  Zimbabwe  ZIM  2012 Summer  2012.0  Summer  London  Swimming   

                                             Event Medal  
227  Swimming Women's 200 metres Individual Medley   NaN

In [ ]:
athletes_filtered.at[227, 'Height']=176.0 #i googled her true height

### 3. Age task

In [ ]:
import heapq

In [ ]:
athletes_age_filt = athletes_filtered[athletes_filtered['Age'].notna()]
males = athletes_age_filt[athletes_age_filt['Sex'] == 'M']
females = athletes_age_filt[athletes_age_filt['Sex'] == 'F']
heapq.nsmallest(2, males['Age'].unique()) # the youngest boys
heapq.nsmallest(2, females['Age'].unique()) # the youngest girls

[11.0, 12.0]

**[11.0, 12.0], [11.0, 12.0]** - for both girls and boys. Thus, the age of the youngest is ***11*** years old.

### 4. Height tasks

In [ ]:
athletes_height_filt = athletes_filtered[athletes_filtered['Height'].notna()]
height_fem = athletes_height_filt[athletes_height_filt['Sex'] == 'F']
height_fem['Height'].mean(), height_fem['Height'].std() # mean height and standard deviation

(167.868536030644, 8.813787841963014)

The mean height of *women* is ***167.87*** with standard deviation ***8.81***.

In [ ]:
height_male = athletes_height_filt[athletes_height_filt['Sex'] == 'M']
height_male['Height'].mean(), height_male['Height'].std() # mean height and standard deviation

(178.90626299944302, 9.366192072615458)

The mean height of *men* is ***178.91*** with standard deviation ***9.37***.

#### 5. Tennis 2000

In [ ]:
tennis_fem = height_fem.loc[(height_fem['Sport'] == 'Tennis') & (height_fem['Year'] == 2000.0)]
round(tennis_fem['Height'].mean(), 1), round(tennis_fem['Height'].std(), 1)

(171.8, 6.5)

The mean height of the female tennisists in 2000 is ***171.8*** with standard deviation ***6.5***

### 6. The heaviest sportsman in 2006

In [ ]:
athletes_2006 = athletes_filtered.loc[(athletes_filtered['Sex'] == 'M') & (athletes_filtered['Year'] == 2006.0)]
heaviest = athletes_2006[athletes_2006['Weight'] == max(athletes_2006['Weight'])]
heaviest['Sport'] # Skeleton!

8102    Skeleton
Name: Sport, dtype: object

The sport of the heaviest person, Patrick R. Antaki, in 2006 is ***skeleton***.

### 7. Golden medals and women

In [ ]:
females_in_range = athletes_filtered.loc[(athletes_filtered['Sex'] == 'F') & (athletes_filtered['Year'].between(1980, 2010)) & (athletes_filtered['Medal'] == 'Gold')]
females_in_range.shape[0] # the number of medales

2245

Women won ***2245 ***medals from 1980 to 2010. 

### 8. John Aalberg

In [ ]:
john = athletes_filtered.loc[athletes_filtered['Name'] == 'John Aalberg']

john.loc[john['Year'] == john['Year'].unique()[0]].shape[0], john.loc[john['Year'] == john['Year'].unique()[1]].shape[0] # in 1992, 1994 respectively

(4, 4)

John Aalberg participated 4 times in winter games in 1992 and the same amount in 1994. 

### 9. Age groups

In [ ]:
athletes_2008 = athletes_filtered.loc[athletes_filtered['Year'] == 2008]
min(athletes_2008['Age']), max(athletes_2008['Age'])

In [ ]:
def age_group(age: int|float) -> str:
    if 20 > age >= 12:
        return '[12; 20)'
    if 35 > age >= 20:
        return '[20; 35)'
    if 50 > age >= 35:
        return '[35; 50)'
    return '[50; 67]'

In [ ]:
ages = athletes_2008['Age']
#athletes_2008['Age group'] = list(map(lambda age: age_group(age), ages))
athletes_2008 = athletes_2008.assign(Age_group = list(map(lambda age: age_group(age), ages)))
athletes_uniq_people = athletes_2008.drop_duplicates(subset=['Name'])
athletes_uniq_people['Age_group'].value_counts()

Thus, **[20; 35)*** most represented group, **[50; 67]** - less represented.

### 10. Comparing 1994 and 2002

In [ ]:
athletes_1994 = athletes_filtered.loc[athletes_filtered['Year'] == 1994]
athletes_2002 = athletes_filtered.loc[athletes_filtered['Year'] == 2002]
athletes_2002['Sport'].unique().shape[0] - athletes_1994['Sport'].unique().shape[0]

More ***3*** sports in 2002!

### 11. Winter and summer medals

In [ ]:
summer_olymp = athletes_filtered.loc[athletes_filtered['Season'] == 'Summer']

gold_summer = summer_olymp.loc[summer_olymp['Medal'] == 'Gold']
silver_summer = summer_olymp.loc[summer_olymp['Medal'] == 'Silver']
bronze_summer = summer_olymp.loc[summer_olymp['Medal'] == 'Bronze']

In [ ]:
gold_summer.value_counts('Team').head(3) # Top 3 countries with golden medal

Team
United States    2363
Soviet Union      830
Germany           568
dtype: int64

Top 3 countries with golden medal in summer games: ***United States***, ***Soviet Union***, ***Germany***. 

In [ ]:
silver_summer.value_counts('Team').head(3) # # Top 3 countries with silver medal

Team
United States    1251
Soviet Union      635
Great Britain     594
dtype: int64

Top 3 countries with silver medal in summer games: ***United States***, ***Soviet Union***, ***Great Britain***. 

In [ ]:
bronze_summer.value_counts('Team').head(3) # # Top 3 countries with bronze medal

Team
United States    1126
Germany           614
Soviet Union      596
dtype: int64

Top 3 countries with bronze medal in summer games: ***United States***, ***Germany***, ***Soviet Union***.

In [ ]:
winter_olymp = athletes_filtered.loc[athletes_filtered['Season'] == 'Winter']

gold_winter = winter_olymp.loc[winter_olymp['Medal'] == 'Gold']
silver_winter = winter_olymp.loc[winter_olymp['Medal'] == 'Silver']
bronze_winter = winter_olymp.loc[winter_olymp['Medal'] == 'Bronze']

In [ ]:
gold_winter.value_counts('Team').head(3) # Top 3 countries with golden medal

Team
Canada           305
Soviet Union     250
United States    166
dtype: int64

Top 3 countries with golden medal in winter games: ***Canada***, ***Soviet Union***, ***United States***.

In [ ]:
silver_winter.value_counts('Team').head(3) # Top 3 countries with silver medal

Team
United States    308
Canada           199
Norway           165
dtype: int64

Top 3 countries with silver medal in winter games: ***United States***, ***Canada***, ***Norway***.

In [ ]:
bronze_winter.value_counts('Team').head(3) # Top 3 countries with bronze medal

Team
Finland          215
Sweden           177
United States    161
dtype: int64

Top 3 countries with bronze medal in winter games: ***Finland***, ***Sweden***, ***United States***.

More diversity in winter games!

### New variables

#### 12. Z-scores

In [ ]:
athletes_filtered = athletes_filtered.assign(Height_z_scores=stats.zscore(athletes_filtered['Height'], nan_policy='omit'))

In [ ]:
athletes_filtered

ID                    Name Sex   Age  Height  Weight         Team  \
0      68370          Helmut Lehmann   M  25.0   178.0    77.0  Switzerland   
1      68371            Jens Lehmann   M  24.0   188.0    82.0      Germany   
2      68371            Jens Lehmann   M  24.0   188.0    82.0      Germany   
3      68371            Jens Lehmann   M  32.0   188.0    82.0      Germany   
4      68371            Jens Lehmann   M  32.0   188.0    82.0      Germany   
...      ...                     ...  ..   ...     ...     ...          ...   
22597  57367  Velik Nikolov Kapsazov   M  21.0   167.0    67.0     Bulgaria   
22598  57367  Velik Nikolov Kapsazov   M  21.0   167.0    67.0     Bulgaria   
22599  57367  Velik Nikolov Kapsazov   M  25.0   167.0    67.0     Bulgaria   
22600  57367  Velik Nikolov Kapsazov   M  25.0   167.0    67.0     Bulgaria   
22601  57367  Velik Nikolov Kapsazov   M  25.0   167.0    67.0     Bulgaria   

       NOC        Games    Year  Season         City       Sport  \
0      SUI  1984 Summer  1984.0  Summer  Los Angeles    Canoeing   
1      GER  1992 Summer  1992.0  Summer    Barcelona     Cycling   
2      GER  1992 Summer  1992.0  Summer    Barcelona     Cycling   
3      GER  2000 Summer  2000.0  Summer       Sydney     Cycling   
4      GER  2000 Summer  2000.0  Summer       Sydney     Cycling   
...    ...          ...     ...     ...          ...         ...   
22597  BUL  1956 Summer  1956.0  Summer    Melbourne  Gymnastics   
22598  BUL  1956 Summer  1956.0  Summer    Melbourne  Gymnastics   
22599  BUL  1960 Summer  1960.0  Summer         Roma  Gymnastics   
22600  BUL  1960 Summer  1960.0  Summer         Roma  Gymnastics   
22601  BUL  1960 Summer  1960.0  Summer         Roma  Gymnastics   

                                                Event   Medal  
0            Canoeing Men's Kayak Fours, 1,000 metres     NaN  
1      Cycling Men's Individual Pursuit, 4,000 metres  Silver  
2            Cycling Men's Team Pursuit, 4,000 metres    Gold  
3      Cycling Men's Individual Pursuit, 4,000 metres  Silver  
4            Cycling Men's Team Pursuit, 4,000 metres    Gold  
...                                               ...     ...  
22597                          Gymnastics Men's Rings     NaN  
22598                Gymnastics Men's Pommelled Horse     NaN  
22599           Gymnastics Men's Individual AllAround     NaN  
22600                 Gymnastics Men's Team AllAround     NaN  
22601                                    Gymnastics M     NaN  

[265893 rows x 15 columns]

#### 13. Min-max normalisation (additional task)

In [ ]:
max_height = max(athletes_filtered['Height']) # 226 is a maximum height
min_height = min(athletes_filtered['Height']) # 127 is a minimum height
delta = max_height - min_height

In [ ]:
athletes_filtered = athletes_filtered.assign(Height_min_max_scaled=athletes_filtered['Height'].apply(
    lambda x: (x - min_height) / delta if x is not np.NaN else np.NaN)
)

### 14. Height, weight, age

In [ ]:
def find_mean(data: pd.DataFrame, sex: str, var: str) -> pd.DataFrame:
    return data.loc[data['Sex'] == sex][var].dropna().mean()

height_women = find_mean(winter_olymp, 'F', 'Height')
weight_women = find_mean(winter_olymp, 'F', 'Weight')
age_women = find_mean(winter_olymp, 'F', 'Age')

height_men = find_mean(winter_olymp, 'M', 'Height')
weight_men = find_mean(winter_olymp, 'M', 'Weight')
age_men = find_mean(winter_olymp, 'M', 'Age')

In [ ]:
data_for_publ = {'Mean height' : [height_men, height_women, height_men - height_women],
                 'Mean weight' : [weight_men, weight_women, weight_men - weight_women],
                 'Mean age' : [age_men, age_women, age_men - age_women]}

comparisons = pd.DataFrame(data_for_publ, index=['Men', 'Women', 'Difference'])

path = 'data/comparisons.xlsx'
comparisons.to_excel(f'{path}', index=True)

#Is excel table enough for publication? Who knows; for such small table i prefer use tables in google docs/word  or in overleaf

### 15. Team and medal

These variables are connected. It is obvious when you see results of top countries with different medals (in summer the united states are always first), and there are always soviet union or some european countries. And we can find the correlation coefficient

In [ ]:
athletes_filtered['Team'].astype('category').cat.codes.corr(athletes_filtered['Medal'].astype('category').cat.codes) #hmmm not big enough after all

0.05997730335402653

In [ ]:
medaliststs = athletes_filtered[~athletes_filtered['Medal'].isna()]
medalists = medaliststs[medaliststs['Team'].isin(list(medaliststs.value_counts('Team').head(10).index.values))]
medaliststs['Team'].astype('category').cat.codes.corr(medaliststs['Medal'].astype('category').cat.codes)

0.014778365636515514

In [ ]:
athletes_filtered['Team'].astype('category').cat.codes.corr(athletes_filtered['Medal'].dropna().astype('category').cat.codes)

0.002181642340942651

In [ ]:
medaliststs['Team'].astype('category').cat.codes.corr(medalists['Medal'].astype('category').cat.codes)

0.012443469091629645